core/desugarRegexes.ml

```diff
@@ -90,7 +90,9 @@ object(self)
           self#phrase (fn_appl libfn tyargs
                             [e1; desugar_regex self#phrase regex_type r])
     | InfixAppl ((_tyargs, BinaryOp.RegexMatch _), _, _) ->
-        raise (Errors.SugarError (pos, "Internal error: unexpected rhs of regex operator"))
+        let (_, expr) = SourceCode.Position.resolve_start_expr pos in
```

Comment:
This looks strange. Can't we recover the RHS expression from the pattern match:

```
InfixAppl ((_tyargs, BinaryOp.RegexMatch _), _, rhs_expr)
```
?

Reply:
Sure, but `expr` here is the string representation from the source code rather than an AST itself, so is arguably more useful to report in the error.

Reply:
Ah, I see.